pkg/auth/handler/gear/templates.go

```diff
@@ -0,0 +1,50 @@
+package gear
+
+import (
+	"encoding/json"
+	"net/http"
+	"sort"
+
+	"github.com/skygeario/skygear-server/pkg/auth"
+	"github.com/skygeario/skygear-server/pkg/core/inject"
+	"github.com/skygeario/skygear-server/pkg/core/server"
+	"github.com/skygeario/skygear-server/pkg/core/template"
+)
+
+func AttachTemplatesHandler(
+	server *server.Server,
+	authDependency auth.DependencyMap,
+) *server.Server {
+	server.
+		Handle("/.skygear/templates.json", &TemplatesHandlerFactory{}).
+		Methods("GET")
+	return server
+}
+
+type TemplatesHandlerFactory struct {
+	Dependency auth.DependencyMap
+}
+
+func (f TemplatesHandlerFactory) NewHandler(request *http.Request) http.Handler {
+	h := &TemplatesHandler{}
+	inject.DefaultRequestInject(h, f.Dependency, request)
+	return h
+}
+
+type TemplatesHandler struct {
+	TemplateEngine *template.Engine `dependency:"TemplateEngine"`
+}
+
+func (h TemplatesHandler) ServeHTTP(rw http.ResponseWriter, r *http.Request) {
+	encoder := json.NewEncoder(rw)
+	rw.Header().Set("Content-Type", "application/json")
+
+	specs := make([]template.Spec, len(h.TemplateEngine.TemplateSpecs))
```

Comment:
The TemplateEngine here has the app tenant config that has ability to query app's templates. Currently it is okay that we are just displaying the template specs which is not app specific information. To avoid we update this handler incorrectly in future, maybe we can choose either:

1. Add comment here to make sure we aware this handler is public, and should not return any templates that related to the app. Or
1. Inject a template engine that has empty tenant config for this handler?

Reply:
In future, the information returned by this handler may indeed depends on the tenant config. For example, we may want to hide templates for disabled features. Therefore, I suggest we keep it as is. Also, at the moment the public interface of `TemplateEngine` cannot be used to access raw tenant config, so there's not much chance to be misused.